dvc/objects/db/reference.py

```diff
@@ -0,0 +1,66 @@
+import io
+import logging
+import os
+from typing import TYPE_CHECKING
+
+from ..errors import ObjectFormatError
+from ..file import ReferenceHashFile
+from .base import ObjectDB
+
+if TYPE_CHECKING:
+    from dvc.fs.base import BaseFileSystem
+    from dvc.hash_info import HashInfo
+    from dvc.types import AnyPath
+
+logger = logging.getLogger(__name__)
+
+
+class ReferenceObjectDB(ObjectDB):
+    """Reference ODB.
+
+    File objects are stored as ReferenceHashFiles which reference paths outside
+    of the staging ODB fs. Tree objects are stored natively.
+    """
+
+    def get(self, hash_info: "HashInfo"):
+        if hash_info.isdir:
+            return super().get(hash_info)
+        path_info = self.hash_to_path(hash_info.value)
```

Comment:
For the record: as we've discussed with @pmrowla before, it might make sense to either store these ref objects in a some subdir in odb (e.g. `refs/` or smth) or maybe use a file extention (similar to `.dir`, e.g. `.ref`) to distinguish these objects from actual real objects.

This might get useful in cases where you need to store actual objects as well as ref objects and is kinda like git's object header, but encoded into a path or filename, which allows us to very easilly distinguish them on sight and through API calls like `list_objects`.